pkg/sdkclient/serverinfo/types.go

```diff
@@ -27,11 +29,72 @@ const (
 
 type sdkConstraints map[Language]string
 
+/*
+minimumSupportedSDKVersions is the minimum supported version of each SDK for the current numaflow version.
+It is used to check if the SDK is compatible with the current numaflow version.
+
+NOTE: when updating it, please also update MINIMUM_SUPPORTED_SDK_VERSIONS for mono vertex at rust/monovertex/server_info.rs
+
+Python SDK versioning follows PEP 440 (https://www.python.org/dev/peps/pep-0440/).
+The other SDKs follow the semver versioning scheme (https://semver.org/).
+
+How to update this map:
+
+There are two types of releases, one is the stable release and the other is the pre-release.
+Below are the typical formats of the versioning scheme:
+
+	+------------------+-------------------------+-----------------------------+
+	|                  |          PEP 440        |            semver           |
+	+------------------+-------------------------+-----------------------------+
+	|      stable      |          0.8.0          |           0.8.0             |
+	+------------------+-------------------------+-----------------------------+
+	|   pre-release    |    0.8.0a1,             |       0.8.0-rc1,            |
+	|                  |    0.8.0b3,             |  0.8.0-0.20240913163521,    |
+	|                  |    or 0.8.0rc1          |            etc.             |
+	+------------------+-------------------------+-----------------------------+
+
+There are two cases to consider when updating the map:
+
+1. The minimum supported version is a pre-release version.
+In this case, directly put the exact pre-release version in the map.
+E.g., if the minimum supported version is "0.8.0-rc1", then put "0.8.0-rc1" for java, go, rust.
+"0.8.0b1", "0.8.0b1" for python.
+2. The minimum supported version is a stable version.
+In this case, put (almost) the largest available pre-release version of the stable version in the map.
+E.g., if the minimum supported version is "0.8.0", then put "0.8.0-z" for java, go, rust, "0.8.0rc100" for python.
```

Comment:
Curious, why using `-z`?

Reply:
Yes, it's mainly because of the golang semver [library](https://github.com/Masterminds/semver?tab=readme-ov-file#checking-version-constraints) "considering pre-releases to be invalid if the ranges does not include one". So we have to put a pre-release version in the constraint. To do so, we need to put the largest pre-release, hence `-z` and `rc100` for python.

I will improve the docs accordingly.